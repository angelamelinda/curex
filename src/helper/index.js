export function filterFloat(value) {
    if (/^(\-|\+)?([0-9]+(\.[0-9]+)?|Infinity)$/
      .test(value))
      return Number(value);
    return NaN;
}
export function isEmpty(obj) {
    for (var key in obj) {
        if (obj.hasOwnProperty(key)) {
        return false;
        }
        return true;
    }
}
export function getCurrencyName(sym) {
    switch (sym) {
        case "USD":
        return "United State Dollar"
        case "CAD":
        return "Canadian Dollar"
        case "IDR":
        return "Indonesia Rupiah"
        case "GBP":
        return "British Pound"
        case "CHF":
        return "Swiss Franc"
        case "SGD":
        return "Singapore Dollar"
        case "INR":
        return "India Rupee"
        case "MYR":
        return "Malaysia Ringgit"
        case "JPY":
        return "Japan Yen"
        case "KRW":
        return "Korea Won"
        default:
        break;
    }
}