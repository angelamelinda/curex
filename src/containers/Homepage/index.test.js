import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import { Homepage } from './';
import * as helper from '../../helper';

Enzyme.configure({ adapter: new Adapter()});

describe('Homepage', () => {
  let props = helper.mockData();
  it('renders correctly', () => {
    // search for the div which shown before the form
    const wrapper = shallow(
        <Homepage {...props} ></Homepage>
    );

    expect(toJson(wrapper),{mode:'shallow'}).toMatchSnapshot();
  });
})
