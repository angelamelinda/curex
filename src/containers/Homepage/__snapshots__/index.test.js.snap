// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage renders correctly 1`] = `
<div
  className="container"
>
  <Connect(Component)
    amountAlert=""
    amountValue="1.00"
    handleChange={[Function]}
  />
  <div
    className="p-sm-4 mb-5"
  >
    <Connect(ListCurrency) />
    <Connect(AddCurrency)
      amountValue="1.00"
    />
  </div>
</div>
`;
